packages/astro/src/core/errors/errors-data.ts

```diff
@@ -1304,6 +1304,77 @@ export const UnknownFilesystemError = {
 	hint: 'It can be caused by many things, eg. missing permissions or a file not existing we attempt to read. Check the error cause for more details.',
 } satisfies ErrorData;
 
+/**
+ * @docs
+ * @description
+ * Cannot extract the font type from the given URL.
+ */
+export const CannotExtractFontType = {
+	name: 'CannotExtractFontType',
+	title: 'Cannot extract the font type from the given URL.',
+	hint: 'Open an issue at https://github.com/withastro/astro/issues',
+} satisfies ErrorData;
+
+/**
+ * @docs
+ * @description
+ * Cannot fetch the given font file
+ * @message
+ * An error occured while fetching font file from the given URL.
+ */
+export const CannotFetchFontFile = {
+	name: 'CannotFetchFontFile',
+	title: 'Cannot fetch the given font file.',
+	message: (url: string) => `An error occured while fetching font file from ${url}`,
+	hint: 'Check the error cause and open an issue if it cannot be solved on your side.',
+} satisfies ErrorData;
+
+/**
+ * @docs
+ * @description
+ * A local font file referenced in your Astro config has been deleted. Restore the file or update your Astro config.
+ */
+export const DeletedLocalFont = {
+	name: 'DeletedLocalFont',
+	title: 'A local font has been deleted',
+	message: 'A local font file referenced in your config has been deleted.',
+	hint: 'Restore the file or update your config.',
+} satisfies ErrorData;
+
+/**
+ * @docs
+ * @description
+ * Cannot load font provider
+ * @message
+ * An error occured while loading the given provider. Open an issue on the corresponding repository.
```

Comment:
```suggestion
 * Astro is unable to load the given font provider. Open an issue on the corresponding provider's repository.
```
I think here, more than just "an error occurred" we might want to more strongly imply "Astro knows what it's doing, but it can't make sense of what it was given." Otherwise, it's hard to know that this isn't an Astro error.